app/uk/gov/hmrc/merchandiseinbaggage/controllers/DeclarationConfirmationController.scala

```diff
@@ -67,9 +67,18 @@ class DeclarationConfirmationController @Inject() (
 
   val makeAnotherDeclaration: Action[AnyContent] = actionProvider.journeyAction.async { implicit request =>
     import request.declarationJourney._
-    repo.upsert(DeclarationJourney(sessionId, declarationType, isAssistedDigital = request.isAssistedDigital)) map {
-      _ =>
-        Redirect(routes.GoodsDestinationController.onPageLoad)
+    val isFromAdminDomain: Boolean = request.headers
+      .get("x-forwarded-host")
+      .exists(host => host.startsWith("admin") || host.startsWith("test-admin"))
+    if (isFromAdminDomain)
+      Future {
+        Redirect(routes.ImportExportChoiceController.onPageLoad)
+      }
+    else {
+      repo.upsert(DeclarationJourney(sessionId, declarationType, isAssistedDigital = request.isAssistedDigital)) map {
+        _ =>
+          Redirect(routes.GoodsDestinationController.onPageLoad)
+      }
```

Comment:
I don't think the functionality should be altered here, as the `upsert` operation is specific to the public-facing mode in this implementation. The ticket indicates that the correct rerouting should be handled for admin users. Additionally, there's no need for `isFromAdminDomain` since `request.isAssistedDigital` is already available. Therefore, the solution can simply be:

```
  val makeAnotherDeclaration: Action[AnyContent] = actionProvider.journeyAction.async { implicit request =>
    import request.declarationJourney._
    repo.upsert(DeclarationJourney(sessionId, declarationType, isAssistedDigital = request.isAssistedDigital)) map {
      _ =>
        if (request.isAssistedDigital) {
          Redirect(routes.ImportExportChoiceController.onPageLoad)
        } else {
          Redirect(routes.GoodsDestinationController.onPageLoad)
        }
    }
  } 
```
So, perform the `upsert` operation and then determine the redirect route based on the value of the flag.